buildSrc/src/main/java/org/elasticsearch/gradle/Jdk.java

```diff
@@ -20,37 +20,41 @@
 package org.elasticsearch.gradle;
 
 import org.gradle.api.Buildable;
-import org.gradle.api.Project;
 import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.model.ObjectFactory;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.TaskDependency;
 
 import java.io.File;
-import java.util.Arrays;
-import java.util.Collections;
 import java.util.Iterator;
 import java.util.List;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
 public class Jdk implements Buildable, Iterable<File> {
 
     private static final List<String> ALLOWED_VENDORS = List.of("adoptopenjdk", "openjdk");
-    static final Pattern VERSION_PATTERN = Pattern.compile("(\\d+)(\\.\\d+\\.\\d+)?\\+(\\d+(?:\\.\\d+)?)(@([a-f0-9]{32}))?");
-    private static final List<String> ALLOWED_PLATFORMS = Collections.unmodifiableList(Arrays.asList("darwin", "linux", "windows", "mac"));
+    private static final List<String> ALLOWED_PLATFORMS = List.of("darwin", "linux", "windows", "mac");
```

Comment:
future note: we should really do the translation for darwin/mac (whichever we consider the oddity) instead of having the user have to know both may exist depending on the version/vendor.

Reply:
Agreed, and this will be much easier to ditch when we move to using the JVM catalog for this.